From 4b4cfcddab5b3d97cd34e750af68cee13b003375 Mon Sep 17 00:00:00 2001
From: Steve Loughran <stevel@apache.org>
Date: Thu, 19 Jan 2017 11:24:58 +0000
Subject: [PATCH 2221/2748] HADOOP-13496. Include file lengths in Mismatch in
 length error for distcp. Contributed by Ted Yu

(cherry picked from commit 77401bd5fcca5127c9908156971eeec468371f47)

Change-Id: Ic47cb65893edcc3b297e7b1a367b229ade42e3bf
---
 .../tools/mapred/RetriableFileCopyCommand.java     |    7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/hadoop-tools/hadoop-distcp/src/main/java/org/apache/hadoop/tools/mapred/RetriableFileCopyCommand.java b/hadoop-tools/hadoop-distcp/src/main/java/org/apache/hadoop/tools/mapred/RetriableFileCopyCommand.java
index 5b88cbb..2571183 100644
--- a/hadoop-tools/hadoop-distcp/src/main/java/org/apache/hadoop/tools/mapred/RetriableFileCopyCommand.java
+++ b/hadoop-tools/hadoop-distcp/src/main/java/org/apache/hadoop/tools/mapred/RetriableFileCopyCommand.java
@@ -190,9 +190,10 @@ private void compareFileLengths(FileStatus sourceFileStatus, Path target,
                                   throws IOException {
     final Path sourcePath = sourceFileStatus.getPath();
     FileSystem fs = sourcePath.getFileSystem(configuration);
-    if (fs.getFileStatus(sourcePath).getLen() != targetLen)
-      throw new IOException("Mismatch in length of source:" + sourcePath
-                + " and target:" + target);
+    long srcLen = fs.getFileStatus(sourcePath).getLen();
+    if (srcLen != targetLen)
+      throw new IOException("Mismatch in length of source:" + sourcePath + " (" + srcLen +
+          ") and target:" + target + " (" + targetLen + ")");
   }
 
   private void compareCheckSums(FileSystem sourceFS, Path source,
-- 
1.7.9.5

